diff --git a/app/models/preview_card.rb b/app/models/preview_card.rb
index a6ec839f8..cbb9c8c31 100644
--- a/app/models/preview_card.rb
+++ b/app/models/preview_card.rb
@@ -27,9 +27,9 @@
 #
 
 class PreviewCard < ApplicationRecord
   IMAGE_MIME_TYPES = ['image/jpeg', 'image/png', 'image/gif'].freeze
-  LIMIT = 1.megabytes
+  LIMIT = 10.megabytes
 
   BLURHASH_OPTIONS = {
     x_comp: 4,
     y_comp: 4,
