diff --git a/etc/nginx/sites-available/mastodon.init b/etc/nginx/sites-available/mastodon
index c1053f6..1b93027 100644
--- a/etc/nginx/sites-available/mastodon.init
+++ b/etc/nginx/sites-available/mastodon
@@ -37,6 +37,14 @@ server {
 }
 
 server {
+  if ($http_user_agent ~* "*.2heng.xin") {
+     return 403;
+  }
+
+  if ($http_user_agent ~* "*.mastodonhub.xin") {
+     return 403;
+  }
+
   listen 443 ssl http2;
   listen [::]:443 ssl http2;
   server_name mastodon.ktachibana.party www.mastodon.ktachibana.party;
@@ -71,12 +79,16 @@ server {
     try_files $uri @proxy;
   }
 
-  location ~ ^/(emoji|packs|system/accounts/avatars|system/media_attachments/files) {
+  location ~ ^/(emoji|packs) {
     add_header Cache-Control "public, max-age=31536000, immutable";
     add_header Strict-Transport-Security "max-age=31536000";
     try_files $uri @proxy;
   }
 
+  location /system {
+    rewrite ^/system(.*) https://mastodon-s3.ktachibana.party$1 permanent;
+  }
+
   location /sw.js {
     add_header Cache-Control "public, max-age=0";
     add_header Strict-Transport-Security "max-age=31536000";
