diff --git a/app/serializers/rest/preview_card_serializer.rb b/app/serializers/rest/preview_card_serializer.rb
index 66ff47d22..08e1b0119 100644
--- a/app/serializers/rest/preview_card_serializer.rb
+++ b/app/serializers/rest/preview_card_serializer.rb
@@ -9,6 +9,10 @@ class REST::PreviewCardSerializer < ActiveModel::Serializer
              :image, :embed_url, :blurhash
 
   def image
-    object.image? ? full_asset_url(object.image.url(:original)) : nil
+    if object.provider_name == 'pixiv'
+      object.embed_url
+    else
+      object.image? ? full_asset_url(object.image.url(:original)) : nil
+    end
   end
 end
