diff --git a/app/models/preview_card.rb b/app/models/preview_card.rb
index 7579178f8..218b6af31 100644
--- a/app/models/preview_card.rb
+++ b/app/models/preview_card.rb
@@ -39,7 +39,7 @@ class PreviewCard < ApplicationRecord
   include Attachmentable
 
   IMAGE_MIME_TYPES = ['image/jpeg', 'image/png', 'image/gif', 'image/webp'].freeze
-  LIMIT = Rails.configuration.x.use_vips ? 8.megabytes : 2.megabytes
+  LIMIT = Rails.configuration.x.use_vips ? 40.megabytes : 10.megabytes
 
   BLURHASH_OPTIONS = {
     x_comp: 4,
