diff --git a/app/javascript/mastodon/features/ui/components/columns_area.js b/app/javascript/mastodon/features/ui/components/columns_area.js
index f4824f045..677132e0d 100644
--- a/app/javascript/mastodon/features/ui/components/columns_area.js
+++ b/app/javascript/mastodon/features/ui/components/columns_area.js
@@ -50,6 +50,7 @@ export default class ColumnsArea extends ImmutablePureComponent {
     isModalOpen: PropTypes.bool.isRequired,
     singleColumn: PropTypes.bool,
     children: PropTypes.node,
+    closeNavigationPanel: PropTypes.bool,
   };
 
   // Corresponds to (max-width: $no-gap-breakpoint + 285px - 1px) in SCSS
@@ -134,8 +135,12 @@ export default class ColumnsArea extends ImmutablePureComponent {
   }
 
   render () {
-    const { columns, children, singleColumn, isModalOpen } = this.props;
+    const { columns, children, singleColumn, isModalOpen, closeNavigationPanel } = this.props;
     const { renderComposePanel } = this.state;
+    let mainPanelStyle = {};
+    if (closeNavigationPanel) {
+      mainPanelStyle = { width: '100%' };
+    }
 
     if (singleColumn) {
       return (
@@ -146,16 +151,18 @@ export default class ColumnsArea extends ImmutablePureComponent {
             </div>
           </div>
 
-          <div className='columns-area__panels__main'>
+          <div className='columns-area__panels__main' style={mainPanelStyle}>
             <div className='tabs-bar__wrapper'><div id='tabs-bar__portal' /></div>
             <div className='columns-area columns-area--mobile'>{children}</div>
           </div>
 
-          <div className='columns-area__panels__pane columns-area__panels__pane--start columns-area__panels__pane--navigational'>
-            <div className='columns-area__panels__pane__inner'>
-              <NavigationPanel />
+          {!closeNavigationPanel &&
+            <div className='columns-area__panels__pane columns-area__panels__pane--start columns-area__panels__pane--navigational'>
+              <div className='columns-area__panels__pane__inner'>
+                <NavigationPanel />
+              </div>
             </div>
-          </div>
+          }
         </div>
       );
     }
diff --git a/app/javascript/mastodon/features/ui/components/header.js b/app/javascript/mastodon/features/ui/components/header.js
index 4e109080e..1c13c6e17 100644
--- a/app/javascript/mastodon/features/ui/components/header.js
+++ b/app/javascript/mastodon/features/ui/components/header.js
@@ -24,11 +24,13 @@ class Header extends React.PureComponent {
 
   static propTypes = {
     location: PropTypes.object,
+    closeNavigationPanel: PropTypes.bool,
+    handleToggleNavigationPanel: PropTypes.func.isRequired,
   };
 
   render () {
     const { signedIn } = this.context.identity;
-    const { location } = this.props;
+    const { location, closeNavigationPanel, handleToggleNavigationPanel } = this.props;
 
     let content;
 
@@ -37,6 +39,12 @@ class Header extends React.PureComponent {
         <>
           {location.pathname !== '/publish' && <Link to='/publish' className='button'><FormattedMessage id='compose_form.publish' defaultMessage='Publish' /></Link>}
           <Account />
+          {/*Derived from $sidebar-width: 55px*/}
+          <div style={{ width: '30px' }} onClick={handleToggleNavigationPanel} role='button' tabIndex={0}>
+            <div style={{ width: '50%', margin: '0 auto' }}>
+              {closeNavigationPanel ? <i className='fa fa-bars' /> : <i className='fa fa-times' />}
+            </div>
+          </div>
         </>
       );
     } else {
diff --git a/app/javascript/mastodon/features/ui/index.js b/app/javascript/mastodon/features/ui/index.js
index b05956606..53ae0539f 100644
--- a/app/javascript/mastodon/features/ui/index.js
+++ b/app/javascript/mastodon/features/ui/index.js
@@ -120,6 +120,7 @@ class SwitchingColumnsArea extends React.PureComponent {
     children: PropTypes.node,
     location: PropTypes.object,
     mobile: PropTypes.bool,
+    closeNavigationPanel: PropTypes.bool,
   };
 
   componentWillMount () {
@@ -150,7 +151,7 @@ class SwitchingColumnsArea extends React.PureComponent {
   }
 
   render () {
-    const { children, mobile } = this.props;
+    const { children, mobile, closeNavigationPanel } = this.props;
     const { signedIn } = this.context.identity;
 
     let redirect;
@@ -170,7 +171,7 @@ class SwitchingColumnsArea extends React.PureComponent {
     }
 
     return (
-      <ColumnsAreaContainer ref={this.setRef} singleColumn={mobile}>
+      <ColumnsAreaContainer ref={this.setRef} singleColumn={mobile} closeNavigationPanel={closeNavigationPanel}>
         <WrappedSwitch>
           {redirect}
 
@@ -254,6 +255,7 @@ class UI extends React.PureComponent {
 
   state = {
     draggingOver: false,
+    closeNavigationPanel: true,
   };
 
   handleBeforeUnload = e => {
@@ -539,8 +541,12 @@ class UI extends React.PureComponent {
     this.context.router.history.push('/follow_requests');
   }
 
+  handleToggleNavigationPanel = () => {
+    this.setState({ closeNavigationPanel: !this.state.closeNavigationPanel });
+  }
+
   render () {
-    const { draggingOver } = this.state;
+    const { draggingOver, closeNavigationPanel } = this.state;
     const { children, isComposing, location, dropdownMenuIsOpen, layout } = this.props;
 
     const handlers = {
@@ -563,14 +569,15 @@ class UI extends React.PureComponent {
       goToBlocked: this.handleHotkeyGoToBlocked,
       goToMuted: this.handleHotkeyGoToMuted,
       goToRequests: this.handleHotkeyGoToRequests,
+      handleToggleNavigationPanel: this.handleToggleNavigationPanel,
     };
 
     return (
       <HotKeys keyMap={keyMap} handlers={handlers} ref={this.setHotkeysRef} attach={window} focused>
         <div className={classNames('ui', { 'is-composing': isComposing })} ref={this.setRef} style={{ pointerEvents: dropdownMenuIsOpen ? 'none' : null }}>
-          <Header />
+          <Header closeNavigationPanel={closeNavigationPanel} handleToggleNavigationPanel={this.handleToggleNavigationPanel} />
 
-          <SwitchingColumnsArea location={location} mobile={layout === 'mobile' || layout === 'single-column'}>
+          <SwitchingColumnsArea location={location} mobile={layout === 'mobile' || layout === 'single-column'} closeNavigationPanel={closeNavigationPanel}>
             {children}
           </SwitchingColumnsArea>
 
